### DISCUSSION TITLE
Modify URL without reloading the page

### DISCUSSION THREAD

**Discussion by kevinNgws** [gatsbyjs/gatsby#32946] [Help]
Hi, is there a way to modify the URL without reloading the page?
My use case for this:
1. Append language to the url
2. Change language whenever user selects another language
3. For example, /en/page to /kr/page

I tried using window.location.replace, but it brings me to 404 page not found

Anyone can help me with this?

Thanks in advance

**Comment by iffjey** [comment]
You probably need to use `navigate` function. It allows you to navigate to pages programmatically without reload. Here's the link to documentation: https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-link/#add-state-to-programmatic-navigation

**Reply by kevinNgws** [reply]
@iffjey I tried with navigate function. It will still reload.
